src/jit/codegencommon.cpp

```diff
@@ -10588,6 +10628,45 @@ void CodeGen::genSetScopeInfoUsingsiScope()
 }
 #endif // USING_SCOPE_INFO
 
+#ifdef USING_VARIABLE_LIVE_RANGE
+void CodeGen::genSetScopeInfoUsingVariableRanges()
+{
+    VariableLiveKeeper* varLiveKeeper = compiler->getVariableLiveKeeper();
+
+    unsigned int varNum;
+    LclVarDsc*   varDsc;
+    unsigned int liveRangeIndex = 0;
+
+    for (varDsc = compiler->lvaTable, varNum = 0; varNum < compiler->info.compLocalsCount; varNum++, varDsc++)
+    {
+        const LiveRangeList* liveRanges = varLiveKeeper->getLiveRangesForVar(varNum);
+
+        if (compiler->compMap2ILvarNum(varNum) != (unsigned int)ICorDebugInfo::UNKNOWN_ILNUM)
+        {
+            for (LiveRangeListIterator itLiveRanges = liveRanges->begin(); itLiveRanges != liveRanges->end();
```

Comment:
`for (const VariableLiveRange& liveRange : liveRanges)` (with appropiate changes to the surrounding code, of course)

Reply:
I like that, thanks!